### ISSUE TITLE
[BUG] [Formatter] right shift in twig files

### ISSUE THREAD

**Issue by palansher** [Riverside-Healthcare/djLint#335]
Good day!

I see a lot of formatting issues for twig. Let's look at one of them:

## System Info
 - OS:  ubuntu 20.04
 - Python Version (``3.8.10``)
 - djLint Version (``1.9.5`)
 - template language: twig

## Issue
on *every* next "--reformat" , the middle line 

```
Карточка` водителя  <b> &emsp;&emsp; {{ (driver is defined and driver is not null) ? '[' ~ driver.getDriverFIO ~ ']': '' }} </b>
```

shifts right, right, right ...

![image](https://user-images.githubusercontent.com/24850253/185087223-068bcef6-3207-4ce7-aff0-b66e8d0ecd5f.png)

the code:
```
{% block page_head %}
    Карточка водителя  <b> &emsp;&emsp; {{ (driver is defined and driver is not null) ? '[' ~ driver.getDriverFIO ~ ']': '' }} </b>
{% endblock page_head %}
```
the test file (I included only the problem lines):
[test_indent1.html.twig.txt](https://github.com/Riverside-Healthcare/djLint/files/9358356/test_indent1.html.twig.txt)

## How To Reproduce
```
djlint --reformat test_driver_form_orphan.html.twig 
```
Thank you for your efforts!

**Comment by palansher** [comment]
The second issue in the same env pls:

## Issue

Twig variables don't keep its indent in HTML hierarchy, and shift to first column:

![image](https://user-images.githubusercontent.com/24850253/185094288-d9db4b26-9242-4e3c-ba56-99f6c07d376b.png)

```
<select class="form-control" name="services_works" {{ topsActive ? '' : 'disabled' }}>
    {% for top in topsActive %}
        <option value="{{ top.id }}">
            {{ top.shortName }}
        </option>
    {% endfor %}

</select>
```

[test_var_shift.html.twig.txt](https://github.com/Riverside-Healthcare/djLint/files/9358714/test_var_shift.html.twig.txt)


my respect!

**Comment by christopherpickering** [comment]
Thanks! It seems that both are triggered by the `--preserve-leading-space` option. I'll check it out.

**Comment by christopherpickering** [comment]
Can you help me understand for what purpose you have the leading space option? It is inteded for non-html files and will make the identing for html files not work correctly.

I'd recommend you update your config to
```
{
"ignore": "H021,H030,H031,H017",
    "extension": "twig",
    "indent": "4",
    "profile": "nunjucks",
    "format_attribute_template_tags": "true",
    "max_line_length": "120",
    "max_attribute_length": "15",
    "preserve_blank_lines": true,
}
```

**Comment by palansher** [comment]
>Can you help me understand for what purpose you have the leading space option? 

Due to my stupidity.

>It is inteded for non-html files and will make the identing for html files not work correctly.

Really, I did not know that. My intention was - to keep my custom manual indents, in case reformat will work undesirable way.

Now formatting is almost perfect! Thank you.

One point, pls:

Currently, in my file, html **tags** indents by 3 spaces (according to "indent": "3" setting).

But the indentation of the html **attributes** of these tags looks sporadically (1, 2 .. indentation)..

I see two problems here:

1) attribute indentation more than one to right from HTML tag indent (namely two+)

2) attribute indentation (2,1 ..) is random and not a multiple of ""indent": "x","

So, I see such a picture in my VS Code:

![image](https://user-images.githubusercontent.com/24850253/185156945-4a4d78a7-68a8-4d5b-a663-438a61620783.png)

![image](https://user-images.githubusercontent.com/24850253/185157052-bde1da28-9493-4c71-acbb-ebad2d24706b.png)

![image](https://user-images.githubusercontent.com/24850253/185157226-451ad265-2884-4c35-b59d-5309fcca0517.png)


Is there a way to fix this? Maybe I did something wrong again.

```
{
    "ignore": "H021,H030,H031,H017",
    "extension": "twig",
    "indent": "3",
    "profile": "nunjucks",
    "format_attribute_template_tags": "true",
    "max_line_length": "120",
    "max_attribute_length": "15",
    "preserve_blank_lines": true    
}
```

[test_driver_form_orphan.html.twig.txt](https://github.com/Riverside-Healthcare/djLint/files/9363501/test_driver_form_orphan.html.twig.txt)

Thank you!

**Comment by palansher** [comment]
Also, maybe you will find that

this formatting
![image](https://user-images.githubusercontent.com/24850253/185168173-5e1ee81a-893c-4bff-bcb1-2cc901ad68b6.png)

is more readable than this:
![image](https://user-images.githubusercontent.com/24850253/185168502-89fc33ed-42c0-42ac-a38a-2c7bdd009092.png)

I mean, that same indent of open (<) and close (>) braces looks more readable for complex hierarchy.

**Comment by christopherpickering** [comment]
Yes, the attributes always indent so the start of the attributes will match. I don't have any plans to change that.

**Comment by palansher** [comment]
> Yes, the attributes always indent so the start of the attributes will match. 

Cool!

But maybe better if  an attributes indents will also pay attention to  "indent" param from config , to keep common indent step ?

**Comment by palansher** [comment]
Thanks for your help!!

**Comment by chasegiunta** [comment]
@christopherpickering I think you already have this change planned, but just want to echo that I'd hope there would be a way to configure it where attributes could wrap similar to prettier...

```
<input
  type="text"
  name="firstName"
  id="first-name"
  autocomplete="given-name"
  class="form-input mt-1"
  value="{{ user.firstName }}"
/>
```
as opposed to the current way
<img width="374" alt="image" src="https://user-images.githubusercontent.com/1377169/219909249-32b825be-bf3b-467e-907b-6d20017d347f.png">

Thank you for considering this and the tremendous work you've done so far!

**Comment by nullbio** [comment]
@christopherpickering Is there a way to disable the auto-wrapping altogether? I have tried setting a large max_line_length, setting it to false, line_break_after_multiline_tag = false, max_attribute_length = 30

For example, this is wrapped starting after the src attribute:

`<img src="{% static 'images/django.png' %}" width="480" height="167" alt="Django logo" />`

I would have thought max_attribute_length was the setting, but doesn't seem to work.
